lm_eval/evaluator.py

```diff
@@ -479,12 +482,14 @@ def evaluate(
                         if "alias" in metrics:
                             metrics.pop("alias")
 
-                        current_size = metrics.pop("samples")
                         # TODO: There should be a way for users
                         #       to toggle between weighted and
                         #       unweighted averaging
-                        # For unweighted averaging, use:
-                        #     current_size = 1
+                        if weight_by_size:
```

Comment:
`weight_by_size` should be set in a benchmark's YAML file IMO. it shouldn't be controllable by CLI at least, that'll create weird discrepancies where one user might set it and another might not, resulting in a divergence in scores that comparing the config YAML files would not reveal

Reply:
That makes sense